File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Features/SetValuePage.vb
Namespace CompanionKit.Controls.ProgressBar.Features

	Public Class SetValuePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        ''' <summary>
        ''' Handles ValueChanged event of the TrackBar that changes of the ProgressBar.
        ''' </summary>
        Private Sub mobjChangevalueTrackBar_ValueChanged(sender As Object, e As EventArgs) Handles mobjChangevalueTrackBar.ValueChanged
            ' Set a new value of the TrackBar for the ProgressBar.
            Me.mobjDemoProgressBar.Value = Me.mobjChangevalueTrackBar.Value
        End Sub

        ''' <summary>
        ''' Handles Load event of the example' UserControl.
        ''' </summary>
        Private Sub SetValuePage_Load(sender As Object, e As EventArgs) Handles Me.Load
            ' Set initial value for TrackBar that changes value for progress bar.
            Me.mobjChangevalueTrackBar.Value = Me.mobjDemoProgressBar.Value
            Me.mobjChangevalueTrackBar.Minimum = Me.mobjDemoProgressBar.Minimum
            Me.mobjChangevalueTrackBar.Maximum = Me.mobjDemoProgressBar.Maximum
            Me.mobjChangevalueTrackBar.TickFrequency = Me.mobjDemoProgressBar.[Step]
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Features/SetValuePage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.ProgressBar.Features
{
    public partial class SetValuePage : UserControl
    {
        public SetValuePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles ValueChanged event of the TrackBar that changes of the ProgressBar.
        /// </summary>
        private void mobjChangevalueTrackBar_ValueChanged(object sender, EventArgs e)
        {
            // Set a new value of the TrackBar for the ProgressBar.
            this.mobjDemoProgressBar.Value = this.mobjChangevalueTrackBar.Value;
        }

        /// <summary>
        /// Handles Load event of the example' UserControl.
        /// </summary>
        private void SetValuePage_Load(object sender, EventArgs e)
        {
            // Set initial value for TrackBar that changes value for progress bar.
            this.mobjChangevalueTrackBar.Value = this.mobjDemoProgressBar.Value;
            this.mobjChangevalueTrackBar.Minimum = this.mobjDemoProgressBar.Minimum;
            this.mobjChangevalueTrackBar.Maximum = this.mobjDemoProgressBar.Maximum;
            this.mobjChangevalueTrackBar.TickFrequency = this.mobjDemoProgressBar.Step;
        }
    }
}